enhancements/machine-config/admin-defined-node-disruption-policy.md

```diff
@@ -0,0 +1,311 @@
+
+As there is a need for this in 4.16, we will proceed with the format that works best with the existing MCO architecture, while leaving it open to change for the future.
+
+#### Hypershift [optional]
+
+As noted above, Hypershift cannot directly tap into this. In a bit more detail, Hypershift has 2 modes of operation for nodepools:
+Replace upgrades - since all machines are deleted and recreated in this model, there is no method of live-application today
+Inplace upgrades - these actually use a special form of the daemon, and we can leverage that to update and perform necessary policies. However we probably do not want hosted cluster admins to have direct control over this, much like MachineConfigs, so if we do want to add this we will need a new API in Hypershift to handle this. As such, we are not considering this at this time
+
+### Risks and Mitigations
+
+As noted in the non-goal sections, any NodeDisruptionPolicy is not verified for application correctness in the MCO, minus formatting. This means that if you accidentally define a change that needed a reboot to take effect, the cluster would “finish the update” without being able to use the new configuration.
+
+The user thereby needs to understand that no additional verification is taking place, and that they accept any responsibility for NodeDisruptionPolicies. If any are applied to the cluster, a metric should be raised for debuggability.
+
+Given the popularity of using pause on MachineConfigPools, it is possible some users will exploit this to skip upgrade disruption. That will likely not break the cluster but will put the configuration in an awkward spot, as we can no longer guarantee that the configuration a node claims to be on is actually applied.
+
+The MCO rebootless apply and reboot apply actually does have 1 major difference still: the application of proper selinux labels. The MCO doesn’t currently have any ability to properly label files it writes, and some users have worked around this by relabelling via a service, which would trigger during the early boot process. With this, we will break that workflow unless the MCO explicitly has the ability to handle selinux labels.
+
+### Drawbacks
+
+There shouldn’t be many drawbacks (this is a pretty lightweight sub-feature that we already in-part ship today) other than what was outlined in Risks and Mitigations.
+
+## Design Details
+
+### Open Questions [optional]
+
+The biggest question is the format and constraints we would like for the UX to be. Conceivably, we have the options to:
+
+ - Have the user specify a global policy 
+ - Allow users to wildcard paths
+ - Allow users to define policies per pool instead of globally
+ - Provide users more flexibility in defining the actions (e.g. provide a shell script to run to change selinux context)
```

Comment:
User can write a systemd unit to achieve that and that unit itself can be defined in the policy :)